### PR TITLE
ci: support different cases of the env vars

### PR DESCRIPTION
**Pull request by spacewander**
Especially for env var GOPROXY, which is the name defined in the Go standard.

Commit Message: ci: support different cases of the env vars
Additional Description:
Risk Level: Low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by phlax — COMMENTED** [review]
@spacewander i dont think this is doing anything

was there a particular issue you were trying to resolve?

**Comment by spacewander** [comment]
First, before this PR, only https_proxy can set up the HTTPS proxy. After this change, both the HTTPS_PROXY and https_proxy can be used, therefore people don't need to remember which env variable should be used. So does the http_proxy/HTTP_PROXY.

Second, this script uses `go_proxy` (instead of `GOPROXY`) env variable to set up the `GOPROXY` env used by docker. So the `GOPROXY` in the user's dev environment doesn't take effect. However, only `GOPROXY` is defined in the Go's official doc: https://go.dev/ref/mod#environment-variables. They have to define another env var `go_proxy` for Envoy's script. This change makes both `GOPROXY` and `go_proxy` work as expected.

File: ci/run_envoy_docker.sh [hunk 1]
```diff
@@ -9,13 +9,21 @@ function is_windows() {
   [[ "$(uname -s)" == *NT* ]]
 }
 
+function export_env() {
+    default="$1"
+    alt="$2"
+
+    if [ -z "${!default}" ]; then
+        export "$default"="$alt"
```
Comment (phlax) [inline ci/run_envoy_docker.sh:17]: how is this any different to just exporting the vars as is removed below?

Comment (phlax) [reply ci/run_envoy_docker.sh:17]: ok i see it only sets it if its not set already

i think we want to make this more explicit - this is not easy to understand what its doing

File: ci/run_envoy_docker.sh [hunk 2]
```diff
@@ -9,13 +9,21 @@ function is_windows() {
...
+}
+
 read -ra ENVOY_DOCKER_OPTIONS <<< "${ENVOY_DOCKER_OPTIONS:-}"
 
-# TODO(phlax): uppercase these env vars
-export HTTP_PROXY="${http_proxy:-}"
-export HTTPS_PROXY="${https_proxy:-}"
-export NO_PROXY="${no_proxy:-}"
-export GOPROXY="${go_proxy:-}"
+export_env HTTP_PROXY "${http_proxy:-}"
```
Comment (phlax) [inline ci/run_envoy_docker.sh:23]:
```suggestion
export HTTP_PROXY "${http_proxy:-${HTTP_PROXY:-}}"
```

Comment (spacewander) [reply ci/run_envoy_docker.sh]: @phlax 
Thanks for your suggestion! Adjusted the change in the new way.

**Review by phlax — APPROVED** [review]
lgtm, thanks @spacewander

**Pull request merged**